sycl/test/on-device/online_compiler/online_compiler_L0.cpp

```diff
@@ -0,0 +1,46 @@
+// REQUIRES: level_zero
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %sycl_source_dir -lze_loader %s -o %t.out
```

Comment:
These flags are needed: "-fsycl-targets=%sycl_triple -I %sycl_source_dir"
For example, sycl_source_dir is required to help compiler to find the header "ze_api.hpp"

Reply:
Is this how users will be required to build their programs? I think -fsycl must still be enough for them, hence for the test too.

Reply:
-fsycl-targets=%sycl_triple% allows to avoid many unnecessary warnings and error in LIT test. Please grep for the -fsycl-targets in the sycl/test directory.
For %sycl_source_dir, yes that is how user need to compile now. It is not problem of this particular pull request of this LIT test. Other tests for ze_loader do the same. For example, https://github.com/intel/llvm/blob/sycl/sycl/test/on-device/plugins/interop-level-zero.cpp#L2

Reply:
`on_device` are supposed to be end-to-end tests, so IMO the closer it is to the user experience the better.

Re %sycl_source_dir - I think it is quite unexpected for the user to require additional options when an Intel extension is used. E.g. to use subgroups no additional includes are needed. @pvchupin, do you have an opinion here?

In any case, I think for the first patch it is OK to require the additional option. We can adjust later, and meanwhile TODO can be added if needed.

Reply:
I agree with @kbobrovs. -I is unexpected. I'm not sure about specifying triple, it is also somewhat unexpected to be, but there could be some valid LIT infra reasons. @vladimirlaz, please comment here.